lib/tar.ml

```diff
@@ -646,65 +686,75 @@ module HeaderReader(Async: ASYNC)(Reader: READER with type 'a t = 'a Async.t) =
   open Reader
 
 
-  let read ?level (ifd: Reader.in_channel) : (Header.t, [ `Eof ]) result t =
+  let read ?level ?global (ifd: Reader.in_channel) : (Header.t * Header.Extended.t option, [ `Eof ]) result t =
     let level = Header.get_level level in
     (* We might need to read 2 headers at once if we encounter a Pax header *)
     let buffer = Cstruct.create Header.length in
     let real_header_buf = Cstruct.create Header.length in
 
-    let next_block () =
+    let next_block global () =
       really_read ifd buffer
       >>= fun () ->
-      return (Header.unmarshal ~level buffer) in
+      match Header.unmarshal ?extended:global ~level buffer with
+      | None -> return None
+      | Some hdr -> return (Some (hdr, global))
+      in
 
-    (* Skip Pax GlobalExtendedHeaders *)
-    let next () =
-      next_block ()
+    let rec next global () =
+      next_block global ()
```

Comment:
I can remove the unit parameter. I've found that using an optional labelled argument for `global` led me to forget it in some callsites.

Reply:
After some thinking I don't think how compatibility levels are handles when reading make much sense (see https://github.com/mirage/ocaml-tar/issues/125). If we get rid of that we can make `~global` a required argument, and the returned updated global state always makes sense to consider.

Reply:
I will leave #125 for another PR :-)